src/assistant/SuperflexAssistant.ts

```diff
@@ -182,6 +184,12 @@ export default class SuperflexAssistant implements Assistant {
       )
     }
 
+    if (!(await isUploadAllowedByTierAndSize(documentPaths))) {
```

Comment:
Put this as constant and then use it. Having await in if statement makes code harder to read